crates/ruff_python_parser/src/string.rs

```diff
@@ -71,14 +71,16 @@ impl StringParser {
         skipped_str
     }
 
+    /// Returns the current position of the parser considering the offset.
     #[inline]
-    fn get_pos(&self) -> TextSize {
-        self.offset + TextSize::try_from(self.cursor).unwrap()
+    fn position(&self) -> TextSize {
+        self.compute_position(self.cursor)
     }
 
+    /// Computes the position of the cursor considering the offset.
     #[inline]
-    fn range(&self, start_location: TextSize) -> TextRange {
-        TextRange::new(start_location, self.offset)
+    fn compute_position(&self, cursor: usize) -> TextSize {
```

Comment:
The mix of `usize` and `TextSize` in this file results in some awkward conversions. It might be good to use `TextSize` consistently throughout the file (doesn't need to be part of this PR)